packages/fyllut/src/components/FormPage.jsx

```diff
@@ -1,48 +1,56 @@
+import {
+  AmplitudeProvider,
+  FyllUtRouter,
+  i18nData,
+  useAppConfig,
+} from "@navikt/skjemadigitalisering-shared-components";
 import React, { useEffect, useState } from "react";
-import { AmplitudeProvider, FyllUtRouter } from "@navikt/skjemadigitalisering-shared-components";
-import { localizationUtils } from "@navikt/skjemadigitalisering-shared-domain";
-
-const { getLanguageCodeAsIso639_1, zipCountryNames } = localizationUtils;
-
-const loadCountryNames = async (locale) => {
-  return fetch(`/fyllut/countries?lang=${getLanguageCodeAsIso639_1(locale)}`).then((response) => response.json());
-};
+import { loadCountryNamesForLanguages, loadFormTranslations, loadGlobalTranslationsForLanguages } from "../api";
 
 function FormPage({ form }) {
   const [translation, setTranslation] = useState({});
-  const [countryNameTranslations, setCountryNameTranslations] = useState({});
+  const [ready, setReady] = useState(false);
+  const { featureToggles } = useAppConfig();
 
   useEffect(() => {
-    fetch(`/fyllut/translations/${form.path}`, { headers: { accept: "application/json" } }).then((response) => {
-      response
-        .json()
-        .then((loadedTranslations) => {
-          setTranslation(loadedTranslations);
-          return loadedTranslations;
-        })
-        .then(async (loadedTranslations) => {
-          const localesInTranslations = Object.keys(loadedTranslations);
-          const countryNamesInBokmaal = await loadCountryNames("nb-NO");
-          await Promise.all(localesInTranslations.map(loadCountryNames)).then((loadedCountryNames) => {
-            const result = loadedCountryNames.reduce(
-              (acc, countryNamesPerLocale, index) => ({
-                ...acc,
-                [localesInTranslations[index]]: zipCountryNames(
-                  countryNamesInBokmaal,
-                  countryNamesPerLocale,
-                  (countryName) => countryName.label
-                ),
-              }),
-              {}
-            );
-            setCountryNameTranslations(result);
-          });
-        });
+    async function fetchTranslations() {
+      if (featureToggles.enableTranslations) {
+        const localTranslationsForForm = await loadFormTranslations(form.path);
+        const availableLanguages = Object.keys(localTranslationsForForm);
+        const countryNameTranslations = await loadCountryNamesForLanguages(availableLanguages);
+        const globalTranslations = await loadGlobalTranslationsForLanguages(availableLanguages);
+
+        return availableLanguages.reduce(
+          (accumulated, lang) => ({
+            ...accumulated,
+            [lang]: {
+              ...accumulated[lang],
+              ...countryNameTranslations[lang],
+              ...globalTranslations[lang],
+              ...localTranslationsForForm[lang],
+            },
+          }),
+          availableLanguages.length > 0 ? { "nb-NO": i18nData["nb-NO"] } : {}
```

Comment:
test om vi skal sende in i18nData her eller ikke, vi trenger i18nData uansett

Reply:
Fikset: 17b3320b1373af42cfb3beaed15f5c18b93453b2

Reply:
Fikses av 17b3320b1373af42cfb3beaed15f5c18b93453b2 og 17b3320b1373af42cfb3beaed15f5c18b93453b2.
Ved å legge til i18nData her vil språkvelgeren i utgangspunktet vises. Har løst det ved å sette en begrensing hvor språkvelgeren i fyllut kun vises når den faktisk har mer enn ett valg.